http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala

```diff
@@ -237,17 +241,29 @@ class ProxyRoutes(proxyService: ProxyService, corsSupport: CorsSupport, refererC
     for {
       ctx <- ev.ask[AppContext]
       apiCall = proxyService.proxyRequest(userInfo, googleProject, runtimeName, request)
-      _ <- metrics.incrementCounter("proxyRuntime")
       resp <- ctx.span.fold(apiCall)(span => spanResource[IO](span, "proxyRuntime").use(_ => apiCall))
+
+      headerMap: Map[String, String] = request.headers.map(header => (header.name(), header.value())).toMap
+      contentLength = Try(headerMap.get("content-length").map(_.toDouble)).toOption.flatten
+
+      _ <- if (request.uri.toString.endsWith(".ipynb") && request.method.equals(Method.PUT)) {
+        contentLength.traverse(size => metrics.gauge("proxy/notebooksSize", size))
+      } else IO.unit
+
+      _ <- if (resp.status.isSuccess()) {
```

Comment:
Should we use this pattern (count successes and failures separately) for other methods (e.g. apps)?

Reply:
oo I like that, seems like a useful thing to collect👍

Reply:
➕